login/profile_blackbox_test.go

```diff
@@ -0,0 +1,164 @@
+package login_test
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+	"reflect"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/almighty/almighty-core/account"
+	"github.com/almighty/almighty-core/auth"
+	config "github.com/almighty/almighty-core/configuration"
+	"github.com/almighty/almighty-core/errors"
+	"github.com/almighty/almighty-core/gormtestsupport"
+	"github.com/almighty/almighty-core/test"
+	"github.com/goadesign/goa"
+	uuid "github.com/satori/go.uuid"
+
+	"github.com/almighty/almighty-core/login"
+
+	"github.com/almighty/almighty-core/resource"
+	_ "github.com/lib/pq"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+)
+
+type ProfileBlackBoxTest struct {
+	gormtestsupport.RemoteTestSuite
+	clean          func()
+	profileService login.UserProfileService
+	configuration  *config.ConfigurationData
+	loginService   *login.KeycloakOAuthProvider
+}
+
+func TestRunProfileBlackBoxTest(t *testing.T) {
+	resource.Require(t, resource.Remote)
+	suite.Run(t, &ProfileBlackBoxTest{RemoteTestSuite: gormtestsupport.NewRemoteTestSuite("../config.yaml")})
+}
+
+// SetupSuite overrides the DBTestSuite's function but calls it before doing anything else
+// The SetupSuite method will run before the tests in the suite are run.
+// It sets up a database connection for all the tests in this suite without polluting global space.
+func (s *ProfileBlackBoxTest) SetupSuite() {
+
+	var err error
+	s.configuration, err = config.GetConfigurationData()
+	if err != nil {
+		panic(fmt.Errorf("Failed to setup the configuration: %s", err.Error()))
+	}
+
+	keycloakUserProfileService := login.NewKeycloakUserProfileClient()
+	s.profileService = keycloakUserProfileService
+
+}
+
+func (s *ProfileBlackBoxTest) SetupTest() {
+}
+
+func (s *ProfileBlackBoxTest) TearDownTest() {
+}
+
+func (s *ProfileBlackBoxTest) generateAccessToken() (*string, error) {
+
+	var scopes []account.Identity
+	scopes = append(scopes, test.TestIdentity)
+	scopes = append(scopes, test.TestObserverIdentity)
+
+	client := &http.Client{Timeout: 10 * time.Second}
+	r := &goa.RequestData{
+		Request: &http.Request{Host: "api.example.org"},
+	}
+	tokenEndpoint, err := s.configuration.GetKeycloakEndpointToken(r)
+
+	assert.Equal(s.T(), tokenEndpoint, "https://sso.prod-preview.openshift.io/auth/realms/fabric8-test/protocol/openid-connect/token")
```

Comment:
Remove this assertion. The endpoint is tested in config tests. If we change this url we don't want to update this tests.